### DISCUSSION TITLE
Update laravel9 to 10

### DISCUSSION THREAD

**Discussion by alexnogueirasilva** [laravel/framework#46124] [General]
Problem 1
    - illuminate/support[v6.0.0, ..., v6.19.1] require php ^7.2 -> your php version (8.2.3) does not satisfy that requirement.
    - illuminate/support[v7.0.0, ..., v7.28.4] require php ^7.2.5 -> your php version (8.2.3) does not satisfy that requirement.
    - illuminate/support[v8.0.0, ..., v8.11.2] require php ^7.3 -> your php version (8.2.3) does not satisfy that requirement.
    - Root composer.json requires laravel/sanctum ^2.15 -> satisfiable by laravel/sanctum[v2.15.0, v2.15.1, 2.x-dev].
    - Conclusion: don't install laravel/framework v10.0.0 (conflict analysis result)
    - laravel/sanctum[v2.15.0, ..., 2.x-dev] require illuminate/support ^6.9|^7.0|^8.0|^9.0 -> satisfiable by illuminate/support[v6.10.0, ..., 6.x-dev, v7.0.0, ..., 7.x-dev, v8.0.0, ..., 8.x-dev, v9.0.0-beta.1, ..., 9.x-dev].
    - Only one of these can be installed: illuminate/support[v6.0.0, ..., 6.x-dev, v7.0.0, ..., 7.x-dev, v8.0.0, ..., 8.x-dev, v9.0.0-beta.1, ..., 9.x-dev, v10.0.0, 10.x-dev], laravel/framework[v10.0.0, 10.x-dev]. laravel/framework replaces illuminate/support and thus cannot coexist with it.
    - Root composer.json requires laravel/framework ^10.0 -> satisfiable by laravel/framework[v10.0.0, 10.x-dev].

when trying to update laravel 9 to 10, i got this error from sanctum, i didn't see anything in the update documentation talking about it, so i set the version of sanctum to plus 3.2.0 so i was able to update

**Comment by kawax** [comment]
sanctum3 should be added to the Laravel10 upgrade guide
https://github.com/laravel/sanctum/blob/3.x/UPGRADE.md

**Comment by Ifriqiya** [comment]
I'm also having an issue, in my case I got this error:
````
  Problem 1
    - illuminate/support[v6.0.0, ..., v6.19.1] require php ^7.2 -> your php version (8.1.1) does not satisfy that requirement.
    - illuminate/support[v7.0.0, ..., v7.28.4] require php ^7.2.5 -> your php version (8.1.1) does not satisfy that requirement.
    - illuminate/support[v8.0.0, ..., v8.11.2] require php ^7.3 -> your php version (8.1.1) does not satisfy that requirement.
    - Root composer.json requires fruitcake/laravel-cors ^3.0.0 -> satisfiable by fruitcake/laravel-cors[v3.0.0].
    - fruitcake/laravel-cors v3.0.0 requires illuminate/support ^6|^7|^8|^9 -> satisfiable by illuminate/support[v6.0.0, ..., v6.20.44, v7.0.0, ..., v7.30.6, v8.0.0, ..., v8.83.27, v9.0.0, ..., v9.52.0].
    - Only one of these can be installed: illuminate/support[v6.0.0, ..., v6.20.44, v7.0.0, ..., v7.30.6, v8.0.0, ..., v8.83.27, v9.0.0, ..., 
v9.52.0, v10.0.0], laravel/framework[v10.0.0]. laravel/framework replaces illuminate/support and thus cannot coexist with it.
    - Root composer.json requires laravel/framework ^10.0.0 -> satisfiable by laravel/framework[v10.0.0].
````

How do I resolve this?

**Reply by alexnogueirasilva** [reply]
maybe the fruitcake/laravel-cors package is causing the error, in my case sanctum 2.x is not supported in laravel 10

**Reply by Ifriqiya** [reply]
It appears to be and that's what I need help with.

**Comment by Ifriqiya** [comment]
Removing fruitcake/laravel-cors resolved this issue. I didn't realise I ought to have removed it since Laravel 9.2

**Comment by Ifriqiya** [comment]
I upgraded to Laravel 10 but I got this error for the phpunit upgrade part:
````
  Problem 1
    - Root composer.json requires pestphp/pest ^1.22.4 -> satisfiable by pestphp/pest[v1.22.4].
    - pestphp/pest v1.22.4 requires nunomaduro/collision ^5.11.0|^6.3.0 -> found nunomaduro/collision[v5.11.0, v6.3.0, v6.3.1, v6.3.2, v6.4.0] but it conflicts with your root composer.json require (^7.0.2).
  Problem 2
    - pestphp/pest v1.22.4 requires nunomaduro/collision ^5.11.0|^6.3.0 -> found nunomaduro/collision[v5.11.0, v6.3.0, v6.3.1, v6.3.2, v6.4.0] but it conflicts with your root composer.json require (^7.0.2).
    - pestphp/pest-plugin-laravel v1.4.0 requires pestphp/pest ^1.22.1 -> satisfiable by pestphp/pest[v1.22.4].
    - Root composer.json requires pestphp/pest-plugin-laravel ^1.4.0 -> satisfiable by pestphp/pest-plugin-laravel[v1.4.0].
````

It appears pestphp 1.22.4 conflicts with nunomaduro/collission 7.0.2 that's required by Laravel 10.

**Reply by alexnogueirasilva** [reply]
do you removed processUncoveredFiles attribute from file phpunit.xml ? and updated the nunomaduro/collision to ^7.0 and phpunit/phpunit to ^10.0 ?

**Reply by Ifriqiya** [reply]
Yes, I did. Removing pestphp sorted the issue.

**Reply by alexnogueirasilva** [reply]
remove the vendor composer.lock folder I think that solves it

**Reply by Ifriqiya** [reply]
OK, thanks for the suggestion.

**Reply by uranov14** [reply]
remove required pestphp/pest v1.22.4 and pestphp/pest-plugin-laravel v1.4.0 from composer.json file.

**Comment by engenes** [comment]
I also got the same error.
Following these steps 

- https://github.com/fruitcake/laravel-cors
- https://github.com/laravel/laravel/pull/5825/files

solved the problem

But I still get the cors error when an error occurs on the server.
For example, if I make a syntax error in the code, causing the application to crash, then the headers will not be set and the browser will show CORS errors.

That is, the execution of ` \Illuminate\Http\Middleware\HandleCors::class` does not work with a 500 error


Installed clean laravel and intentionally made a mistake in the code, and again I see the error cors, instead of an error message

![image](https://user-images.githubusercontent.com/38880660/222970274-9f5fe1fd-b11b-4358-9f76-a9f2c64c73fd.png)
